tests/loops/test_loops.py

```diff
@@ -506,19 +601,35 @@ def configure_optimizers_multiple(self):
     # need to remove these elements for comparison; comparing with `fit_loop.state_dict()` would require the
     # fit loop to have an iterator, which is only available during training
     checkpoint["loops"]["fit_loop"]["state_dict"]["dataloader_state_dict"] = ANY
-
     assert state_dict == checkpoint["loops"]["fit_loop"]
 
-    trainer.fit_loop.load_state_dict(checkpoint["loops"]["fit_loop"])
+    # with `restart_progress=True`, we expect all `ready` counters to be reset to `completed`
```

Comment:
side note: I discussed with @tchaton this morning that `restart_progress` shouldn't be on `load_state_dict`.

And instead we load exactly what was passed, and then each loop is responsible of resetting its state.

Reply:
Yes, I will make a PR for this.